components/org.wso2.carbon.identity.query.saml/src/main/java/org/wso2/carbon/identity/query/saml/SignKeyDataHolder.java

```diff
@@ -109,18 +111,16 @@ public SignKeyDataHolder(String tenantDomain) throws IdentitySAML2QueryException
                 }
 
             } else {
-                keyAlias = ServerConfiguration.getInstance().getFirstProperty(
-                        SECURITY_KEY_STORE_KEY_ALIAS);
+                keyAlias = ServerConfiguration.getInstance().getFirstProperty(SECURITY_KEY_STORE_KEY_ALIAS);
                 if (StringUtils.isBlank(keyAlias)) {
-                    throw new IdentityException("Invalid security configurations in the carbon.xml," +
-                            " The keyAlias is not found for the KeyStore of the tenant domain:" + tenantDomain);
+                    throw new IdentityException("Invalid security configurations in the carbon.xml,"
+                            + " The keyAlias is not found for the KeyStore of the tenant domain:" + tenantDomain);
                 }
 
-                keyAdmin = new KeyStoreAdmin(tenantID,
-                        SAMLSSOUtil.getRegistryService().getGovernanceSystemRegistry());
+                keyAdmin = new KeyStoreAdmin(tenantID, SAMLSSOUtil.getRegistryService().getGovernanceSystemRegistry());
                 keyMan = KeyStoreManager.getInstance(tenantID);
 
-                issuerPK = (PrivateKey) keyAdmin.getPrivateKey(keyAlias, true);
+                issuerPK = (PrivateKey) ServiceReferenceHolder.getKeyProvider().getPrivateKey(tenantDomain);
```

Comment:
getPrivateKey() should accept the key alias rather than the tenant domain since tenant domain might not be the alias all the time.